src/app/pages/my-request-detail/my-request-detail.component.ts

```diff
@@ -0,0 +1,412 @@
+import { MatDialogRef } from '@angular/material/dialog';
+import { ActivatedRoute } from '@angular/router';
+import {
+  CatalogueItemDomainType,
+  DataModelDetail,
+  Uuid,
+} from '@maurodatamapper/mdm-resources';
+import { ToastrService } from 'ngx-toastr';
+import {
+  catchError,
+  EMPTY,
+  filter,
+  finalize,
+  forkJoin,
+  Observable,
+  of,
+  switchMap,
+} from 'rxjs';
+import { BroadcastService } from 'src/app/core/broadcast.service';
+import { DataExplorerService } from 'src/app/data-explorer/data-explorer.service';
+import {
+  DataElementDeleteEvent,
+  DataElementDto,
+  DataElementInstance,
+  DataElementMultipleOperationResult,
+  DataElementOperationResult,
+  DataElementSearchResult,
+  DataRequest,
+  mapToDataRequest,
+  SelectableDataElementSearchResultCheckedEvent,
+} from 'src/app/data-explorer/data-explorer.types';
+import {
+  DataAccessRequestsSourceTargetIntersections,
+  DataRequestsService,
+} from 'src/app/data-explorer/data-requests.service';
+import { DialogService } from 'src/app/data-explorer/dialog.service';
+import {
+  OkCancelDialogData,
+  OkCancelDialogResponse,
+} from 'src/app/data-explorer/ok-cancel-dialog/ok-cancel-dialog.component';
+import { DataModelService } from 'src/app/mauro/data-model.service';
+import { ResearchPluginService } from 'src/app/mauro/research-plugin.service';
+import { RequestElementAddDeleteEvent } from 'src/app/shared/data-element-in-request/data-element-in-request.component';
+
+@Component({
+  selector: 'mdm-my-request-detail',
+  templateUrl: './my-request-detail.component.html',
+  styleUrls: ['./my-request-detail.component.scss'],
+})
+export class MyRequestDetailComponent implements OnInit {
+  request?: DataRequest;
+  requestElements: DataElementSearchResult[] = [];
+  state: 'idle' | 'loading' = 'idle';
+  sourceTargetIntersections: DataAccessRequestsSourceTargetIntersections;
+  removeSelectedButtonDisabled = true;
+
+  constructor(
+    private route: ActivatedRoute,
+    private dataRequests: DataRequestsService,
+    private explorer: DataExplorerService,
+    private toastr: ToastrService,
+    private dataModels: DataModelService,
+    private researchPlugin: ResearchPluginService,
+    private dialogs: DialogService,
+    private broadcast: BroadcastService
+  ) {
+    this.sourceTargetIntersections = {
+      dataAccessRequests: [],
+      sourceTargetIntersections: [],
+    };
+  }
+  ngOnInit(): void {
+    this.initialiseRequest();
+  }
+
+  initialiseRequest(): void {
+    this.route.params
+      .pipe(
+        switchMap((params) => {
+          const requestId = params.requestId;
+          this.state = 'loading';
+          return this.dataRequests.get(requestId);
+        }),
+        catchError((error) => {
+          this.toastr.error(`Invalid Request Id. ${error}`);
+          this.state = 'idle';
+          return EMPTY;
+        })
+      )
+      .subscribe((request) => {
+        this.setRequest(request);
+        this.state = 'idle';
```

Comment:
I'd suggest putting this state change in a `finalize()` operator in the `pipe()`. It will always be called at the end of the pipe (but before the "next" operation in the `subscribe()`).